src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1377 @@
+            }
+            break;
+
+        case GT_NOP:
+            // A GT_NOP is either a passthrough (if it is void, or if it has
+            // a child), but must be considered to produce a dummy value if it
+            // has a type but no child.
+            srcCount = 0;
+            if (tree->TypeGet() != TYP_VOID && tree->gtGetOp1() == nullptr)
+            {
+                assert(dstCount == 1);
+                BuildDef(tree);
+            }
+            else
+            {
+                assert(dstCount == 0);
+            }
+            break;
+
+        case GT_KEEPALIVE:
+            assert(dstCount == 0);
+            srcCount = BuildOperandUses(tree->gtGetOp1());
+            break;
+
+        case GT_JTRUE:
+            srcCount = 0;
+            assert(dstCount == 0);
+            break;
+
+        case GT_JMP:
+            srcCount = 0;
+            assert(dstCount == 0);
+            break;
+
+        case GT_SWITCH:
+            // This should never occur since switch nodes must not be visible at this
+            // point in the JIT.
+            srcCount = 0;
+            noway_assert(!"Switch must be lowered at this point");
+            break;
+
+        case GT_JMPTABLE:
+            srcCount = 0;
+            assert(dstCount == 1);
+            BuildDef(tree);
+            break;
+
+        case GT_SWITCH_TABLE:
+            buildInternalIntRegisterDefForNode(tree);
+            srcCount = BuildBinaryUses(tree->AsOp());
+            assert(dstCount == 0);
+            break;
+
+        case GT_ASG:
+            noway_assert(!"We should never hit any assignment operator in lowering");
+            srcCount = 0;
+            break;
+
+        case GT_ADD:
+        case GT_SUB:
+            if (varTypeIsFloating(tree->TypeGet()))
+            {
+                // overflow operations aren't supported on float/double types.
+                assert(!tree->gtOverflow());
+
+                // No implicit conversions at this stage as the expectation is that
+                // everything is made explicit by adding casts.
+                assert(tree->gtGetOp1()->TypeGet() == tree->gtGetOp2()->TypeGet());
+            }
+            FALLTHROUGH;
+
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+        case GT_LSH:
+        case GT_RSH:
+        case GT_RSZ:
+        case GT_ROR:
+            srcCount = BuildBinaryUses(tree->AsOp());
+            buildInternalRegisterUses();
+            assert(dstCount == 1);
+            BuildDef(tree);
+            break;
+
+        case GT_RETURNTRAP:
+            // this just turns into a compare of its child with an int
+            // + a conditional call
+            BuildUse(tree->gtGetOp1());
+            srcCount = 1;
+            assert(dstCount == 0);
+            killMask = compiler->compHelperCallKillSet(CORINFO_HELP_STOP_FOR_GC);
+            BuildDefsWithKills(tree, 0, RBM_NONE, killMask);
+            break;
+
+        case GT_MUL:
```

Comment:
There is a `tree->gtOverflow()` case missing under `GT_MUL`. You need a register to check for overflow.

Reply:
OK, thanks